feg/gateway/services/n7_n40_proxy/servicers/create_session_test.go

```diff
@@ -0,0 +1,336 @@
+/*
+Copyright 2021 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package servicers_test
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+
+	mockPolicyDB "magma/feg/gateway/policydb/mocks"
+	n7_sbi "magma/feg/gateway/sbi/specs/TS29512NpcfSMPolicyControl"
+	sbi "magma/feg/gateway/sbi/specs/TS29571CommonData"
+	"magma/feg/gateway/services/n7_n40_proxy/n7"
+	mockN7 "magma/feg/gateway/services/n7_n40_proxy/n7/mocks"
+	"magma/feg/gateway/services/n7_n40_proxy/servicers"
+	"magma/lte/cloud/go/protos"
+)
+
+const (
+	URL1            = "https://mockpcf/npcf-smpolicycontrol/v1"
+	TOKEN_URL       = "https://mockpcf/oauth2/token"
+	CLIENT_ID       = "feg_magma_client"
+	CLIENT_SECRET   = "feg_magma_secret"
+	LOCAL_ADDR      = "127.0.0.1:10100"
+	NOTIFY_API_ROOT = "https://magma-feg.magam.com/npcf-smpolicycontrol/v1"
+	IMSI1           = "IMSI123456789012345"
+	IMSI2           = "IMSI123456789012346"
+	IMSI1_NOPREFIX  = "123456789012345"
+	IMSI2_NOPREFIX  = "123456789012346"
+	UE_IPV4         = "10.1.2.3"
+	PDU_SESSION_ID  = 10
+	GPSI1           = "9876543210"
+	APN1            = "apn.magma.com"
+)
+
+var (
+	AccessType3gpp     = sbi.AccessType("3GPP_ACCESS")
+	DnnMagma           = sbi.Dnn(APN1)
+	Gpsi1              = sbi.Gpsi(GPSI1)
+	UeIpv4             = sbi.Ipv4Addr(UE_IPV4)
+	PduSessionTypeIpv4 = sbi.PduSessionType("IPV4")
+	RatTypeNR          = sbi.RatType("NR")
+	UeTzIST            = sbi.TimeZone("+05:30")
+	SmPolicyUrl        = "https://example.com//npcf-smpolicycontrol/v1/sm-policies/12345"
+	ActTime            = time.Unix(1634906551, 0)
+	DeactTime          = time.Unix(1634913751, 0)
+)
+
+func TestCreateSession(t *testing.T) {
+	srv, mockDb, mockN7 := newCentralSessionControllerForTest(t, false)
+
+	mockN7.On("PostSmPoliciesWithResponse", mock.Anything,
+		mock.MatchedBy(matchSmPolicyContextData(IMSI1_NOPREFIX)),
+	).Return(newSmPolicyResponse(t), nil).Once()
+
+	// no omnipresent rules
+	mockDb.On("GetOmnipresentRules").Return([]string{}, []string{}).Once()
+	// mockDb.On("GetRuleIDsForBaseNames", []string{"base_10"}).Return([]string{"base_rule_1", "base_rule_2"})
```

Comment:
To save test cases, you can test all those rules here. You can add omni rules, static and dynamic and see if the create session looks right